source/include/drivers/cs_Uicr.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Mar 17, 2022
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <protocol/cs_UicrPacket.h>
+#include <protocol/cs_Typedefs.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * Get the hardware board from UICR, or the default board if none is set in UICR.
+ */
+uint32_t getHardwareBoard();
+
+/**
+ * Write the default hardware board to UICR if none is written yet..
+ *
+ * In the past, the hardware board was not written to UICR. Instead, the board was compiled into the firmware,
+ * and each product had their own different firmware.
+ * This function is made to transition from that solution to a single firmware with board in UICR.
+ * We only have to make sure that each product once runs a firmware that writes the correct hardware board to UICR..
+ */
+void writeHardwareBoard();
```

Comment:
Same. Hardware board "type"? A bit more precise language will make it easier to understand.